component/metrics/receiver.go

```diff
@@ -0,0 +1,19 @@
+package metrics
+
+import (
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// Receiver is used to pass an array of metrics to another receiver
+type Receiver struct {
+	// metrics should be considered immutable
+	Receive func(timestamp int64, metrics []FlowMetric) `hcl:"receiver"`
```

Comment:
This is the func that needs to be set by the child component, and the parent calls Receive. The timestamp can apply to all the metrics. We may need to return a mapping of reference ids to flowmetric(s) in the short term.